warehouse/oidc/forms/_core.py

```diff
@@ -30,10 +30,21 @@ class PendingPublisherMixin:
     def validate_project_name(self, field):
         project_name = field.data
 
-        if project_name in self._project_factory:
-            raise wtforms.validators.ValidationError(
-                _("This project name is already in use")
+        try:
+            project = self._project_factory[project_name]
+        except KeyError:
+            # If the project doesn't exist, we're ok to proceed
+            return
+
+        if self._current_user in project.owners:
+            error_msg = _(
+                "Project already exists, create an ordinary "
+                "trusted publisher instead"
             )
```

Comment:
Error message LGTM, but it's a bit unfortunate that we have to plumb the current user all the way into the form logic like this.

Thinking about this a bit: maybe rather than specializing on `user in project.owners`, we can just unconditionally spit out this message. That'll make it slightly wrong for users who don't have permission on the requested project, but that should be a pretty rare case anyways (and will avoid the need to check the current user).

Reply:
Sure, I don't mind either way. That would turn this PR into a one-liner more or less.

Does anyone else have opinions on it?